xdsl/dialects/qssa.py

```diff
@@ -29,12 +30,35 @@ class QubitAttr(ParametrizedAttribute, TypeAttribute):
 qubit = QubitAttr()
 
 
-class QubitBase(IRDLOperation, ABC):
-    pass
+class QssaBase(IRDLOperation, ABC):
+    """
+    Base class for qssa operations, with methods to help convert to the qref dialect.
+
+    Invariant:
+    self.is_gate == self.ref_op().is_gate
+    """
+
+    @abstractmethod
+    def ref_op(self) -> qref.QRefBase:
+        """
+        Build corresponding qref operation
+        """
+        raise NotImplementedError()
+
+    @property
+    @abstractmethod
+    def is_gate(self) -> bool:
+        """
+        Is this operation a gate?
+        Qssa gates represent standard quantum logic gates
+        They should have an equal number of qubit-typed operands and results
+        Their results must be rewired when converting to the qref dialect
```

Comment:
nit:
```suggestion
        Is this operation a gate?
        Qssa gates represent standard quantum logic gates.
        They should have an equal number of qubit-typed operands and results.
        Their results must be rewired when converting to the qref dialect.
```
And in other doc comments